sdk/servicebus/service-bus/samples-v1/typescript/src/receiveMessagesStreaming.ts

```diff
@@ -24,29 +30,48 @@ export async function main() {
   // If receiving from a Subscription, use `createSubscriptionClient` instead of `createQueueClient`
   const queueClient = sbClient.createQueueClient(queueName);
 
-  // To receive messages from sessions, use getSessionReceiver instead of getReceiver or look at
+  // To receive messages from sessions, use createSessionReceiver instead of createReceiver or look at
   // the sample in sessions.ts file
-  const receiver = queueClient.createReceiver(ReceiveMode.peekLock);
 
-  const onMessageHandler: OnMessage = async (brokeredMessage) => {
-    console.log(`Received message: ${brokeredMessage.body}`);
-    await brokeredMessage.complete();
-  };
-  const onErrorHandler: OnError = (err) => {
-    console.log("Error occurred: ", err);
-  };
+  // controls whether we continue to recover from fatal Receiver failures.
+  let enableReceiverRecovery = true;
 
-  try {
-    receiver.registerMessageHandler(onMessageHandler, onErrorHandler, { autoComplete: false });
+  do {
+    const receiver = queueClient.createReceiver(ReceiveMode.peekLock);
 
-    // Waiting long enough before closing the receiver to receive messages
-    await delay(5000);
+    const receiverPromise = new Promise((resolve, _reject) => {
+      const onMessageHandler: OnMessage = async (brokeredMessage) => {
+       
+        console.log(`Received message: ${brokeredMessage.body}`);
+        await brokeredMessage.complete();
+      };
 
+      const onErrorHandler: OnError = (err) => {
+        if (!(err as MessagingError).retryable) {
```

Comment:
Say our attempts to auto complete fails with a retryable error. In that case user's error handler will be called and they will create a new receiver but for no good reason...

Reply:
```suggestion
        if ((err as MessagingError).retryable === false) {
```
I think this will fix at least one problem - if there is an error that doesn't even concern itself with retryability we shouldn't restart.

Reply:
Now the other question is - are the errors that come back from autocomplete failure 'non-retryable' errors. I'll check that out and see.

Reply:
We have an interesting customer case that's triggering this exact condition - I think this is a perfect fit for it.

In particular - the user is experiencing a connection disruption when the receiver is being created. As part of that we then end up in a situation where we exceed our timeouts and the receiver gets an error (is retryable though) and stops receiving.

So this technique causes it to properly handle that, provide an opportunity to log, and restart. Seems solid enough that we should feel comfortable releasing this particular sample.